chat-plugins/poll.js

```diff
@@ -0,0 +1,155 @@
+/*
+* Poll chat plugin
+* This plugin allows roomauth (default: moderator and above) to run a poll in a room.
+* Every room can have one poll, and every user can vote. The results are displayed for
+* users that have voted, and are updated in real time. The poll can be closed with /endpoll.
+* By bumbadadabum with (a lot of) help from Zarel.
+*/
+
+var Poll = (function () {
+    function Poll(room, question, options) {
+        this.room = room;
+        this.question = question;
+        this.voters = [];
+        this.totalVotes = 0;
+
+        this.options = [];
+        for (var i in options) {
+            this.options.push({name: options[i], votes: 0});
+        }
+    }
+
+    Poll.prototype.vote = function (user, option) {
+        for (var ip in user.ips) {
+            if (this.voters.indexOf(ip) > -1) {
+                return this.room.send('You have already voted for this poll.');
+            }
+            else {
+                this.voters.push(ip);
+            }
+        }
+
+        this.options[option - 1].votes++;
+        this.totalVotes++;
+
+        this.update();
+    };
+
+    Poll.prototype.generateVotes = function () {
+        var output = '<div class="infobox" style="text-align:center"><p style="font-weight:bold;font-size:14pt">' + this.question + '</p>';
+        for (var i in this.options) {
+            output += '<button value="/votepoll ' + (parseInt(i) + 1) + '" name="send">' + (parseInt(i) + 1) + '. ' + this.options[i].name + '</button><br/>';
```

Comment:
If you were iterating the options array correctly, you could just do ``(i + 1)``